stdlib/2/__builtin__.pyi

```diff
@@ -1374,19 +1379,106 @@ if sys.version_info >= (3,):
         errors: Optional[str] = ...,
         newline: Optional[str] = ...,
         closefd: bool = ...,
-        opener: Optional[Callable[[str, int], int]] = ...,
-    ) -> TextIO: ...
+        opener: Optional[_Opener] = ...,
+    ) -> TextIOWrapper: ...
+
+    # Unbuffered binary mode: returns a FileIO
     @overload
     def open(
         file: _OpenFile,
         mode: _OpenBinaryMode,
-        buffering: int = ...,
+        buffering: Literal[0],
         encoding: None = ...,
         errors: None = ...,
         newline: None = ...,
         closefd: bool = ...,
-        opener: Optional[Callable[[str, int], int]] = ...,
+        opener: Optional[_Opener] = ...,
+    ) -> FileIO: ...
+
+    # Buffering is on: return BufferedRandom, BufferedReader, or BufferedWriter
+    @overload
+    def open(
+        file: _OpenFile,
+        mode: _OpenBinaryModeUpdating,
+        buffering: Literal[-1, 1],
+        encoding: None = ...,
+        errors: None = ...,
+        newline: None = ...,
+        closefd: bool = ...,
+        opener: Optional[_Opener] = ...,
+    ) -> BufferedRandom: ...
+    @overload
+    def open(
+        file: _OpenFile,
+        mode: _OpenBinaryModeWriting,
+        buffering: Literal[-1, 1],
+        encoding: None = ...,
+        errors: None = ...,
+        newline: None = ...,
+        closefd: bool = ...,
+        opener: Optional[_Opener] = ...,
+    ) -> BufferedWriter: ...
+    @overload
+    def open(
+        file: _OpenFile,
+        mode: _OpenBinaryModeReading,
+        buffering: Literal[-1, 1],
+        encoding: None = ...,
+        errors: None = ...,
+        newline: None = ...,
+        closefd: bool = ...,
+        opener: Optional[_Opener] = ...,
+    ) -> BufferedReader: ...
+
+    # Buffering argument is not given: return BufferedRandom, BufferedReader, or BufferedWriter
```

Comment:
Can't these three overloads not be merged into the overloads above using `...`?

Reply:
Yes, you're right